units/event_send.go

```diff
@@ -0,0 +1,442 @@
+}
+
+type eventNotificationJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	settings *evergreen.Settings
+
+	NotificationID string `bson:"notification_id" json:"notification_id" yaml:"notification_id"`
+}
+
+func makeEventNotificationJob() *eventNotificationJob {
+	j := &eventNotificationJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    eventNotificationJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func newEventNotificationJob(id string) amboy.Job {
+	j := makeEventNotificationJob()
+	j.NotificationID = id
+
+	j.SetID(fmt.Sprintf("%s:%s", eventNotificationJobName, id))
+	return j
+}
+
+func (j *eventNotificationJob) Run(_ context.Context) {
+	defer j.MarkComplete()
+
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		j.AddError(errors.Wrap(err, "error retrieving admin settings"))
+
+	} else if flags == nil {
+		j.AddError(errors.Wrap(err, "fetched no service flags configuration"))
+	}
+	if j.HasErrors() {
+		return
+	}
+
+	if len(j.NotificationID) == 0 {
+		j.AddError(errors.New("notification ID is not valid"))
+		return
+	}
+
+	if j.settings == nil {
+		j.settings, err = evergreen.GetConfig()
+		j.AddError(err)
+		if err == nil && j.settings == nil {
+			j.AddError(errors.New("settings object is nil"))
+		}
+		if j.HasErrors() {
+			return
+		}
+	}
+
+	n, err := notification.Find(j.NotificationID)
+	j.AddError(err)
+	if err == nil && n == nil {
+		j.AddError(errors.Errorf("can't find notification with ID: '%s", j.NotificationID))
+	}
+	if j.HasErrors() {
+		return
+	}
+
+	var sendError error
+	switch n.Subscriber.Type {
+	case event.GithubPullRequestSubscriberType:
+		if err = checkFlag(flags.GithubStatusAPIDisabled); err != nil {
+			j.AddError(err)
+			j.AddError(n.MarkError(err))
+			return
+		}
+
+		sendError = errors.New("unimplemented")
+
+	case event.SlackSubscriberType:
+		if err = checkFlag(flags.SlackNotificationsDisabled); err != nil {
+			j.AddError(err)
+			j.AddError(n.MarkError(err))
+			return
+		}
+		sendError = j.slackMessage(n)
+
+	case event.JIRAIssueSubscriberType:
+		if err = checkFlag(flags.JIRANotificationsDisabled); err != nil {
+			j.AddError(err)
+			j.AddError(n.MarkError(err))
+			return
+		}
+		sendError = j.jiraIssue(n)
+
+	case event.JIRACommentSubscriberType:
+		if err = checkFlag(flags.JIRANotificationsDisabled); err != nil {
+			j.AddError(err)
+			j.AddError(n.MarkError(err))
+			return
+		}
+		sendError = j.jiraComment(n)
+
+	case event.EvergreenWebhookSubscriberType:
+		if err = checkFlag(flags.WebhookNotificationsDisabled); err != nil {
+			j.AddError(err)
+			j.AddError(n.MarkError(err))
+			return
+		}
+		sendError = j.evergreenWebhook(n)
+
+	case event.EmailSubscriberType:
+		if err = checkFlag(flags.EmailNotificationsDisabled); err != nil {
+			j.AddError(err)
+			j.AddError(n.MarkError(err))
+			return
+		}
+		sendError = j.email(n)
```

Comment:
I don't see any value in breaking this up any further it already is broken up

Reply:
I think it makes sense to have these methods on the notification type so that all of the handling is there, rather than in the units package (which I think of as being more higherlevel general.)

Reply:
I thought you made these changes? am I missing a diff?